Revisit the "Developer Framework" getting started section (with bonus tests section rework) (#169)
validmind/documentation: site/guide/testing-overview.qmd

    - ../notebooks/how_to/configure_parameters_demo.ipynb
    - ../notebooks/how_to/implementing_custom_tests.ipynb
  - id: tests-explore
    type: grid
    max-description-length: 250
    sort: false
    fields: [title, description]
    contents:
    - ../notebooks/how_to/explore_tests.ipynb
    - ../notebooks/how_to/explore_test_suites.ipynb
    - test-sandbox.qmd

  - id: tests-intermediate
    type: grid
    max-description-length: 250
    sort: false
    fields: [title, description]
    contents: 
    - ../notebooks/code_samples/custom_tests/external_test_providers_demo.ipynb
    - ../notebooks/how_to/configure_dataset_features.ipynb
    - ../notebooks/how_to/run_documentation_sections.ipynb
    - ../notebooks/how_to/run_documentation_with_config.ipynb
    - ../notebooks/how_to/run_tests_that_require_multiple_datasets.ipynb
    - ../notebooks/how_to/run_unit_metrics.ipynb
  - id: tests-advanced
    type: grid
    max-description-length: 250
    sort: false
    fields: [title, description]
    contents: 
    - ../notebooks/how_to/document_multiple_results_for_the_same_test.ipynb
    - ../notebooks/how_to/load_datasets_predictions.ipynb
---

> ValidMind provides many built-in tests and test suites which make it easy for a model developer to document their work at any point during the model development lifecycle when they need to validate that their work satisfies model risk management requirements. 

Reviewer: ```suggestion
ValidMind provides a variety of built-in tests and test suites, which help you produce documentation during stages of the model development lifecycle where you need to validate that your work satisfies MRM (model risk management) requirements.
```

Author: Removed "a variety of" to simplify text, committed the rest.

> ## Key concepts

Author: Another one to consider including in our glossary: `decorator`

As per John, "It’s a simpler way for users to run their own code as a ValidMind test. Decorators are a pretty common feature in python code so it should be familiar to dev framework users but I think it might be good to include it with a link to some more documentation like this:https://www.datacamp.com/tutorial/decorators-python"

Author: To be addressed in https://app.shortcut.com/validmind/story/4074/create-modular-glossary

- `summary()`: Generates output tables summarizing the results.

<!---
```python
class ExampleMetric(Metric):

        name = "mean_of_values"

        # Markdown-compatible description of the metric
        def description(self):

        # Compute the metric and cache its results and figures
        def run (self):

        # Code to build a list of ResultSummaries that form the results tables
        def summary (self, metric_values)
```
--->

Test suite
: A collection of tests which are run together to automate generating model documentation end-to-end for specific use-cases. 

   Example: the [`classifier_full_suite`](../validmind/validmind/test_suites/classifier.html#ClassifierFullSuite) test suite runs the tests from the [`tabular_dataset`](../validmind/validmind/test_suites/tabular_datasets.html) and [`classifier`](../validmind/validmind/test_suites/classifier.html) test suites to fully document the data and model sections for binary classification model use-cases. 

Test providers
: Third-party test libraries that can be be used with the ValidMind Developer Framework.

## When do I use tests and test suites?

> As a model developer, you always have the flexibility to decide when to use which ValidMind tests. To help you get started, we have identified a few typical scenarios with their own characteristics and needs:

Reviewer: ```suggestion
While you have the flexibility to decide when to use which ValidMind tests, we have identified a few typical scenarios with their own characteristics and needs:
```
